### PR TITLE
Der Freischutz stops skipping leg day and can now walk (and gun) (with knuckles)

### PR DESCRIPTION
**Pull request by TaculoTaculo**
## About The Pull Request
This does not affect the main gamemode at all.
Edits Der Freischutsz for gamemodes where Breach Effect is called, he can now shoot his gun at a target, creating a laser indicator and firing a magic bullet projectile after a delay.
He is also able to zoom his view out and see through walls.

## Why It's Good For The Game
Lets us add a gigantic fan favorite to LCL and RCorp.

## Changelog
:cl:
add: Added new functionality for freischutz on rcorp
balance: adds der frei as an easy support for rcorp
imageadd: added freischutz 96x64 icon
imagedel: deleted freischutz 64x64 icon
refactor: refactored how frei changes icons
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm [hunk 1]
```diff
@@ -23,6 +35,122 @@
...
+/datum/action/innate/abnormality_attack/toggle/der_freischutz_zoom/New(Target)
+	..()
+	original_sight = owner.sight
+
+/datum/action/innate/abnormality_attack/toggle/der_freischutz_zoom/Activate()
+	ActivateSignals()
+	owner.sight |= SEE_TURFS | SEE_MOBS | SEE_THRU
+	owner.regenerate_icons()
+	owner.client.view_size.zoomOut(zoom_out_amt, zoom_amt, owner.dir)
+	. = ..()
```
Comment (Gboster-0) [inline code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm:67]: When `. = ..()` is at the end of a proc, it should be `return ..()` instead
```suggestion
	return ..()
```

File: code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm [hunk 2]
```diff
@@ -23,6 +35,122 @@
...
+
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(Deactivate))
+	RegisterSignal(owner, COMSIG_ATOM_DIR_CHANGE, PROC_REF(Rotate))
+
+/datum/action/innate/abnormality_attack/toggle/der_freischutz_zoom/Deactivate()
+	DeactivateSignals()
+	owner.sight = original_sight
+	owner.regenerate_icons()
+	owner.client.view_size.zoomIn()
+	. = ..()
```
Comment (Gboster-0) [inline code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm:80]: When `. = ..()` is at the end of a proc, it should be `return ..()` instead
```suggestion
	return ..()
```

File: code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm [hunk 3]
```diff
@@ -23,6 +35,122 @@
...
+/mob/living/simple_animal/hostile/abnormality/der_freischutz/proc/IconChange(firing)
+	if(firing)
+		pixel_x -= 32
+		icon = 'ModularTegustation/Teguicons/96x64.dmi'
+		update_icon()
+		return
+	pixel_x += 32
+	icon = 'ModularTegustation/Teguicons/32x64.dmi'
+	update_icon()
+	return
```
Comment (Gboster-0) [inline code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm:113]: No need to return at the end of code
```suggestion
```

**Review by Gboster-0 — COMMENTED** [review]
le review

File: code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm (hunk 3 above)
Comment (Teo-Giambarba) [reply code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm]: Adressed in newest commit.

File: code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm (hunk 2 above)
Comment (Teo-Giambarba) [reply code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm]: Adressed in newest commit.

File: code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm (hunk 1 above)
Comment (Teo-Giambarba) [reply code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm]: Adressed in newest commit.

**Review by Kitsunemitsu — APPROVED** [review]
Approved. I saw this in action, and nothing here looks out of place

**Pull request merged**